routing/router.go

```diff
@@ -734,10 +734,17 @@ func (r *ChannelRouter) pruneZombieChans() error {
 		log.Tracef("Pruning zombie channel with ChannelPoint(%v)",
 			chanToPrune)
 
-		err := r.cfg.Graph.DeleteChannelEdge(&chanToPrune)
+		err := r.cfg.Graph.MarkEdgeZombie(chanToPrune.ChannelID)
+		if err != nil {
+			return fmt.Errorf("unable to mark ChannelPoint(%v) "+
+				"as zombie: %v", chanToPrune.ChannelPoint, err)
+		}
+
+		err = r.cfg.Graph.DeleteChannelEdge(&chanToPrune.ChannelPoint)
```

Comment:
Perhaps we should combine both operations into a single call? If we do, then this means that someone can't give us a super old channel and make us re-scan the graph with GetUTXO or the like.

Reply:
yeah the more i think about this, i think its a good idea to have the transition from live -> zombie be atomic

Reply:
Good point! Fixed.